Missionframework/modules/24_enemy/fnc/fn_enemy_transferGarrison.sqf

```diff
@@ -0,0 +1,174 @@
+    Date: 2019-02-18
+    Last Update: 2019-02-24
+    License: GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.html
+
+    Description:
+        Transfer given amount of infantry or vehicle classes from one sector to another.
+        Given the condition, that the amount of infantry or the specific classname is available at the origin sector.
+        The origin sector also has to be inactive.
+
+        !NOTE! Maybe this could be moved to the garrison module.
+
+    Parameter(s):
+        _origin         - Sector from which the troops should be transfered                     [STRING, defaults to ""]
+        _destination    - Sector to which the troops should be transfered                       [STRING, defaults to ""]
+        _infantry       - Amount of infantry which should be transfered                         [NUMBER, defaults to 0]
+        _vehicleClasses - Array of vehicle classnames which should be transfered                [ARRAY, defaults to []]
+        _onFoot         - True if soldiers should walk instead of being transported by a truck  [BOOL, defaults to false]
+
+    Returns:
+        Array of created groups [ARRAY]
+*/
+
+params [
+    ["_origin", "", [""]],
+    ["_destination", "", [""]],
+    ["_troops", 0, [0]],
+    ["_vehicleClasses", [], [[]]],
+    ["_onFoot", false, [false]]
+];
+
+// Exit, if no origin or destination is given
+if (_origin isEqualTo "" || _destination isEqualTo "") exitWith {[]};
+
+// Exit, if origin sector is currently active
+if (_origin in KPLIB_garrison_active) exitWith {[]};
+
+if (KPLIB_param_debug) then {diag_log format ["[KP LIBERATION] [ENEMY] Transferring garrison from %1 (%2) to %3 (%4) with [%5, %6]", markerText _origin, _origin, markerText _destination, _destination, _troops, _vehicleClasses];};
+
+["KPLIB_transferConvoy_start", [_origin, _destination]] call CBA_fnc_localEvent;
+
+// Arrays for created units
+private _infantry = [];
+private _vehicles = [];
+private _groups = [];
+
+// Origin garrison reference
+private _garrison = [_origin] call KPLIB_fnc_garrison_getGarrison;
+
+// Spawn Pos and Dir function
+private _getPosDir = {
+    params [
+        ["_pos", [], [[]]],
+        ["_radius", 150, [0]]
+    ];
+
+    private _nearRoad = selectRandom (_pos nearRoads _radius);
+    private _spawnPos = (_pos getPos [(random _radius), random 360]) findEmptyPosition [10, 50, "B_T_VTOL_01_armed_F"];
+    private _spawnDir = random 360;
+    if !(isNil "_nearRoad") then {
+        _spawnPos = getPosATL _nearRoad;
+        while {!((_spawnPos nearEntities 20) isEqualTo [])} do {
+            _nearRoad = selectRandom (_pos nearRoads _radius);
+            _spawnPos = getPosATL _nearRoad;
+        };
+        _spawnDir = _nearRoad getDir ((roadsConnectedTo _nearRoad) select 0);
+    };
+
+    [_spawnPos, _spawnDir]
+};
+
+// Handle requested infantry
+if !(_troops > (_garrison select 2)) then {
+    // Define one infantry squad for transport
+    private _squadComp = [
```

Comment:
We should provide some more immersive groups in a later stage.

Leader
Medic
AA/AT
MG
...

Reply:
This is just for moving amount of infantry from A to B.
For "battlegroups" or similar it'll be dynamic like in the old framework.